Slicing overview

HACO8888 opened the pull request:
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

# Checklist:

- [✓] My code follows the style guidelines of this project
- [✓] I have performed a self-review of my code
- [✓] I have commented my code, particularly in hard-to-understand areas
- [✓] I have made corresponding changes to the documentation
- [✓] My changes generate no new warnings
- [✓] I have added tests that prove my fix is effective or that my feature works
- [✓] New and existing unit tests pass locally with my changes

HACO8888 committed:
Update

SiongSng committed:
Update MainView.vue

HACO8888 committed:
Update

HACO8888 committed:
Update PageOverview.vue

SiongSng committed:
fix: recent activities css and other css

HACO8888 committed:
Update

HACO8888 committed:
Update

HACO8888 committed:
Update

HACO8888 committed:
Update

HACO8888 committed:
Update

HACO8888 committed:
Update

HACO8888 committed:
Update RecentBar.vue

SiongSng committed:
fix: search bar

SiongSng committed:
remove `.vscode`

SiongSng committed:
Update NavBar.vue

HACO8888 committed:
Fixed NavBar And Recent Bar

HACO8888 committed:
Update

HACO8888 committed:
Update

HACO8888 committed:
Delete NavBar.vue

HACO8888 committed:
Update

SnowFireWolf committed:
feat: complete theme system

SnowFireWolf committed:
Merge branch 'feature/overview' of https://github.com/Lipoic/Lipoic-Frontend into feature/overview

SnowFireWolf committed:
feat: sidebar active animation

HACO8888 committed:
Update FooBar.vue

HACO8888 committed:
Update

HACO8888 committed:
Update SideBar.vue

HACO8888 committed:
Update RecentBar.vue

SiongSng committed:
style: format code

SiongSng committed:
style: format more code

SiongSng committed:
fix: eslint

SiongSng committed:
style: format tailwindcss

SiongSng committed:
refactor: get user info on mobile content bar

SiongSng commented on package.json:
```diff
@@ -7,13 +7,13 @@
     "coverage": "vitest run --coverage"
   },
   "dependencies": {
+    "@material/web": "^0.1.0-alpha.0",
```
沒有用到的套件可以刪除

SiongSng commented on src/stores/models/settings.ts:
```diff
@@ -0,0 +1,50 @@
+    },
+
+    setTheme(theme: string) {
+      console.log('theme', theme);
```
應該不用 console.log 主題吧

SiongSng committed:
update

SiongSng committed:
style: remove unused background

SiongSng commented:
目前主要還缺單元測試、手機端的 nav bar 動畫、還有手機端的淺色主題

SnowFireWolf committed:
remove @material/web, remove settings store console

SnowFireWolf committed:
feat: complete mobile layout and theme system, and rename

SnowFireWolf committed:
rename component

SiongSng committed:
fix: eslint

SiongSng committed:
fix: mobile card color

SiongSng committed:
fix theme color

SiongSng committed:
fix: dark theme mobile text color

SiongSng commented:
目前應該就單元測試與手機端的上課通知

SiongSng commented on src/components/overview/OverviewLayout.vue:
```diff
@@ -0,0 +1,18 @@
+    <desktop-view />
+  </div>
+
+  <div class="hidden w-[100%] phone:inline-block">
```
或許這裡的佈局邏輯可以更一致一點

上方是如果是手機端就隱藏
下方則是預設隱藏，如果是手機端就顯示

SiongSng commented on src/components/overview/desktop/AppBar.vue:
```diff
@@ -0,0 +1,119 @@
+        />
+      </div>
+    </form>
+    <div class="mt-7 flex-1 text-right">
```
如果切換主題這部份有 tooltip 應該可以讓使用者更了解這些按鈕的用途

SiongSng commented on src/components/overview/desktop/AppBar.vue:
```diff
@@ -0,0 +1,119 @@
+          />
+        </svg>
+      </button>
+      <button id="account" class="mx-[48px]">
```
目前後端已經實做了頭貼 API，所以這裡可以串接使用者實際的頭貼，如果沒有頭貼才換成這個 icon

SiongSng replied:
@MRHACO  將預計挪到另一個分支處理

SiongSng commented on src/components/overview/desktop/RecentCard.vue:
```diff
@@ -0,0 +1,47 @@
+
+<template>
+  <div class="card w-full max-w-[520px]">
+    <img :src="'src/assets/overview/' + props.name + '.png'" />
```
這裡或許可以用 `import()` 來處理圖片路徑的部份

HACO8888 replied:
import 必須在最頂層
宣告props後即無法使用

SiongSng commented on src/components/overview/desktop/SideBar.vue:
```diff
@@ -0,0 +1,250 @@
+
+const navbarItems = [
+  {
+    icon: `
```
這些導航列的 icon 應該可以讓手機端與桌機共用，就不用重複寫相同的程式碼

SiongSng commented on src/components/overview/mobile/AppBar.vue:
```diff
@@ -0,0 +1,76 @@
+          />
+        </svg>
+      </button>
+      <button id="account" class="mr-5 block">
```
這裡與桌面端的串接使用者頭貼 API 相同

SiongSng commented on src/stores/models/settings.ts:
```diff
@@ -0,0 +1,50 @@
+import { defineStore } from 'pinia';
+
+export interface ISettings {
+  theme: string;
```
建議這裡可以弄 `enum` 枚舉這類的東西去定義

SiongSng reviewed (changes requested):
我剛才稍微掃了一下程式碼，發現的幾個問題

HACO8888 committed:
Fixed: Margin Problem

HACO8888 committed:
Update: Read Description

Fix and Change Components Path
Fix View Path

Make Desktop and Mobile Components different path

HACO8888 committed:
Update src/components/Desktop/Overview/AppBar.vue

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

HACO8888 committed:
Update src/components/Mobile/Overview/ContentBar.vue

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

HACO8888 committed:
Update src/components/Mobile/Overview/ContentBar.vue

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

HACO8888 committed:
Update src/components/Mobile/Overview/ContentBar.vue

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

HACO8888 committed:
Update src/components/Mobile/Overview/ContentBar.vue

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

HACO8888 committed:
Update src/components/Desktop/Overview/RecentCard.vue

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

HACO8888 committed:
Update: Add Tooltip for theme mode button

SiongSng committed:
Update src/components/Desktop/Overview/AppBar.vue

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

HACO8888 committed:
Update component

Move public components to main file

HACO8888 committed:
Update RecentCard.vue

HACO8888 committed:
Update

SiongSng committed:
Update src/components/Mobile/AppBar.vue

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SiongSng committed:
Update src/components/Desktop/AppBar.vue

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SiongSng committed:
Update src/components/Mobile/AppBar.vue

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SiongSng committed:
fix: login page styles

SiongSng committed:
refactor: theme

SiongSng committed:
fix: light theme

SiongSng committed:
refactor: navigation bar

SiongSng committed:
Merge branch 'develop' into feature/overview

SiongSng committed:
Merge branch 'develop' into feature/overview

SiongSng committed:
test: desktop recent card

SiongSng reviewed (changes requested):
由於是切版，大多實際功能尚未實做
因此暫時不須單元測試使用者介面的部份，但有部份邏輯上仍需要測試，像是 settings store
我等下會幫忙寫一下#

SiongSng committed:
test: theme settings

SiongSng commented on src/locales/zh-TW.json:
```diff
@@ -1,6 +1,9 @@
   "app": {
     "title": "Lipoic - 與我們一起翻轉教育"
+  },
+  "overview": {
```
這裡目前沒有使用到 應該先不用加進去吧

SiongSng committed:
Update zh-TW.json

SiongSng committed:
fix: recent card dynamic image

SiongSng committed:
Update AppBar.vue

SiongSng reviewed (approved):
LGTM
